app/src/features/surveys/view/SurveyAnimals.tsx

```diff
@@ -70,13 +85,47 @@ const SurveyAnimals: React.FC = () => {
   };
 
   const DeviceFormValues: IAnimalTelemetryDevice = {
-    device_id: 0,
+    device_id: '' as unknown as number,
     device_make: '',
-    frequency: 0,
+    frequency: '' as unknown as number,
     frequency_unit: '',
     device_model: '',
-    attachment_start: '',
-    attachment_end: undefined
+    deployments: [
+      {
+        deployment_id: '',
+        attachment_start: '',
+        attachment_end: undefined
+      }
+    ]
+  };
+
+  const obtainDeviceFormInitialValues = (mode: TelemetryDeviceFormMode) => {
+    switch (mode) {
+      case 'add':
+        return [DeviceFormValues];
+      case 'edit': {
+        const deployments = deploymentData?.filter((a) => a.critter_id === currentCritterbaseCritterId);
+        if (deployments) {
+          //Any suggestions on something better than this reduce is welcome.
```

Comment:
These comments should probably be at the top of the method, describing what it does